projector-core/src/Projector/Core/Prelude.hs

```diff
@@ -0,0 +1,110 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+module Projector.Core.Prelude (
+    module X
+  , fromMaybeM
+  , whenM
+  , unlessM
+  , with
+  , bind
+  , valueOrEmpty
+  , emptyOrValue
+  , orEmpty
+  , maybeToLeft
+  , maybeToRight
+  , leftToMaybe
+  , rightToMaybe
+  , mcase
+  , ecase
+  , renderIntegral
+  , head
+  ) where
+
+import           Projector.Core.Prelude.EitherT as X
+import           Control.Applicative as X
+import           Control.Monad as X
+import           Control.Monad.Trans.Bifunctor as X (BifunctorTrans (..))
+import           Data.Bits as X (Bits (..))
+import           Data.Bool as X (Bool (..), (||), (&&), not, bool, otherwise)
+import           Data.Char as X (Char)
+import           Data.Bifunctor as X (Bifunctor (..))
+import           Data.Either as X (Either (..), either, isLeft, isRight)
+import           Data.Foldable as X
+import           Data.Function as X ((.), ($), (&), flip, id, const, on, fix)
+import           Data.Functor as X (($>))
+import           Data.Int as X
+import           Data.List as X (filter, reverse, find, sort)
+import           Data.Maybe as X (Maybe (..), isNothing, isJust, maybe, fromMaybe, catMaybes, listToMaybe)
+import           Data.Monoid as X (Monoid (..), (<>))
+import           Data.Text as X (Text)
+import qualified Data.Text as Text
+import           Data.Traversable as X
+import           Data.Tuple as X (uncurry)
+import           Data.Word as X (Word8, Word16, Word32, Word64)
+import           Prelude as X (Eq (..), Show (..), Ordering (..), Ord (..), Num (..), Enum, Bounded (..), Integral (..), Double, error, seq, fromIntegral, (/), (^), fst, snd, Integer, Real (..), floor, subtract, fromRational, ($!))
+import           Text.Read as X (Read (..), readMaybe)
```

Comment:
This can probably be culled further - i pulled it out of another project I have and just added things that were used from p for now.